### PR TITLE
Send operation requests to GCP in batch tasks and wait for the instances as last step

### PR DESCRIPTION
**Pull request by keyki**

### CODE REVIEW

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ResourceType.java [hunk 1]
```diff
@@ -64,4 +66,8 @@
     public CommonResourceType getCommonResourceType() {
         return commonResourceType;
     }
+
+    public static boolean isInstanceResource(ResourceType resourceType) {
+        return List.of(GCP_INSTANCE, OPENSTACK_INSTANCE, MOCK_INSTANCE).contains(resourceType);
```
Comment (bergerdenes) [inline cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ResourceType.java]: make the list private static final

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpNetworkInterfaceProvider.java [hunk 2]
```diff
@@ -76,7 +76,7 @@ private String getInstanceNamePrefix(List<CloudResource> instances) {
                 request.setPageToken(response.getNextPageToken());
             } while (response.getNextPageToken() != null);
         } catch (IOException e) {
-            LOGGER.debug("Error dunging metadata collection", e);
+            LOGGER.debug("Error dunging instance collection", e);
```
Comment (bergerdenes) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpNetworkInterfaceProvider.java]: during?

Comment (keyki) [reply cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpNetworkInterfaceProvider.java]: It's not mine, but I'll fix it.

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ResourceType.java (hunk 1 above)
Comment (keyki) [reply cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ResourceType.java]: This is an enum class. Moving the list to a new class is not ideal I think.

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpNetworkInterfaceProvider.java (hunk 2 above)
Comment (keyki) [reply cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpNetworkInterfaceProvider.java]: Fixed.

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ResourceType.java (hunk 1 above)
Comment (keyki) [reply cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ResourceType.java]: Fixed.

**Pull request merged**